plugins/uefi-capsule/fu-uefi-capsule-plugin.c

```diff
@@ -1146,6 +1147,67 @@ fu_uefi_capsule_plugin_coldplug(FuPlugin *plugin, FuProgress *progress, GError *
 	return TRUE;
 }
 
+static gboolean
+fu_uefi_capsule_plugin_cleanup_esp(FuUefiCapsulePlugin *self, GError **error)
+{
+	g_autofree gchar *esp_path = fu_volume_get_mount_point(self->esp);
+	g_autofree gchar *pattern = NULL;
+	g_autoptr(FuDeviceLocker) esp_locker = NULL;
+	g_autoptr(GPtrArray) files = NULL;
+
+	/* delete any files matching the glob in the ESP */
+	esp_locker = fu_volume_locker(self->esp, error);
+	if (esp_locker == NULL)
+		return FALSE;
+	files = fu_path_get_files(esp_path, error);
+	if (files == NULL)
+		return FALSE;
+	pattern = g_build_filename(esp_path, "EFI/*/fw/fwupd*.cap", NULL);
+	for (guint i = 0; i < files->len; i++) {
+		const gchar *fn = g_ptr_array_index(files, i);
+		if (g_pattern_match_simple(pattern, fn)) {
+			g_autoptr(GFile) file = g_file_new_for_path(fn);
+			g_debug("deleting %s", fn);
+			if (!g_file_delete(file, NULL, error))
+				return FALSE;
+		}
+	}
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_uefi_capsule_plugin_reboot_cleanup(FuPlugin *plugin, FuDevice *device, GError **error)
+{
+	FuUefiCapsulePlugin *self = FU_UEFI_CAPSULE_PLUGIN(plugin);
+
+	/* provide an escape hatch for debugging */
+	if (!fu_plugin_get_config_value_boolean(plugin,
+						"RebootCleanup",
+						FU_UEFI_CAPSULE_CONFIG_DEFAULT_REBOOT_CLEANUP))
+		return TRUE;
+
+	/* delete capsules */
+	if (!fu_uefi_capsule_plugin_cleanup_esp(self, error))
+		return FALSE;
+
+	/* delete any old variables */
+	if (!fu_efivar_delete_with_glob(FU_EFIVAR_GUID_FWUPDATE, "fwupd*-*", error))
+		return FALSE;
+
+	/* this should not be required, but, hey -- here were are */
+	if (fu_efivar_exists(FU_EFIVAR_GUID_EFI_GLOBAL, "BootNext")) {
```

Comment:
Need to be careful here. Fwupd isn't the only thing that uses BootNext. I think it needs to only be deleted if the variable points to the Linux Firmware Updater entry.

Reply:
Agree; fixed.